src/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -100,6 +104,8 @@ public function __construct($query, $parserResult, array $queryComponents)
         $this->platform = $query->getEntityManager()->getConnection()->getDatabasePlatform();
         $this->rsm      = $parserResult->getResultSetMapping();
 
+        $query = clone $query;
```

Comment:
how did this end up here?

Reply:
It makes sure that the `Query` object that is passed into the method does not have its `firstResult`/`maxResults` values modified at the caller's side... which is important because...